common/src/main/java/tachyon/worker/block/io/LocalFileBlockWriter.java

```diff
@@ -67,7 +56,12 @@ public WritableByteChannel getChannel() {
 
   @Override
   public long append(ByteBuffer inputBuf) throws IOException {
-    return write(mLocalFileChannel.size(), inputBuf);
+    return write(mLocalFileChannel.size(), inputBuf.duplicate());
```

Comment:
also `duplicate` is not really copying new data but only duplicate the position on the shared data.